### PR TITLE
MACRO-1266 Do not create Redlines for Insert/Delete comments

### PR DESCRIPTION
**Pull request by whutchinson98**
- Removes `comment added` and `comment deleted` from being generated in redlines

https://www.loom.com/share/4914aba666ee410eabba35b20fb2048d?sid=fedb62f5-e200-4fa4-b3e9-fbb382f4ef19

### CODE REVIEW

File: libreoffice-core/sw/source/core/doc/DocumentRedlineManager.cxx [hunk 1]
```diff
@@ -1274,6 +1274,13 @@ DocumentRedlineManager::AppendRedline(SwRangeRedline* pNewRedl, bool const bCall
         return AppendResult::IGNORED;
     }
 
+    // MACRO-1266 - Do not create a delete redline for a comment
```
Comment (skececi) [inline libreoffice-core/sw/source/core/doc/DocumentRedlineManager.cxx:1277]: I like this idea a lot

Comment (chase) [reply libreoffice-core/sw/source/core/doc/DocumentRedlineManager.cxx:1277]: Good call, also lets us see at a glance if something was introduce by us vs. upstream. Will be handy for merge conflicts going forward.

File: libreoffice-core/sw/source/core/doc/DocumentRedlineManager.cxx [hunk 2]
```diff
@@ -1274,6 +1274,13 @@ DocumentRedlineManager::AppendRedline(SwRangeRedline* pNewRedl, bool const bCall
...
+    if(pNewRedl->IsAnnotation()) {
+        delete pNewRedl;
+        pNewRedl = nullptr;
```
Comment (chase) [inline libreoffice-core/sw/source/core/doc/DocumentRedlineManager.cxx:1280]: It's pretty much never a good idea to manually delete a class.
Assigning pNewRdl to nullptr is sufficient.

File: libreoffice-core/sw/source/core/doc/DocumentContentOperationsManager.cxx [hunk 3]
```diff
@@ -4120,9 +4120,7 @@ bool DocumentContentOperationsManager::DeleteAndJoinWithRedlineImpl(SwPaM & rPam
     {
         auto pRedline(std::make_unique<SwRangeRedline>(RedlineType::Delete, rPam));
         if (pRedline->HasValidRange())
-        {
             redlines.push_back(std::move(pRedline));
-        }
```
Comment (chase) [inline libreoffice-core/sw/source/core/doc/DocumentContentOperationsManager.cxx:4125]: It seems like your changes for removing the Insert comment got removed in your latest force push.

Also prefer use { } for `if` `else`, makes reading the code a lot more straight forward.

**Review by chase — CHANGES_REQUESTED** [review]

File: libreoffice-core/sw/source/core/doc/DocumentContentOperationsManager.cxx (hunk 3 above)
Comment (whutchinson98) [reply libreoffice-core/sw/source/core/doc/DocumentContentOperationsManager.cxx:4125]: I don't believe those changes were necessary with the changes I made to the other file. I'll triple check though

File: libreoffice-core/sw/source/core/doc/DocumentRedlineManager.cxx (hunk 2 above)
Comment (chase) [reply libreoffice-core/sw/source/core/doc/DocumentRedlineManager.cxx:1280]: Didn't see that, in that case, do what they do! Definitely makes me curious why, but better go with the flow than against it haha

File: libreoffice-core/sw/source/core/doc/DocumentContentOperationsManager.cxx (hunk 3 above)
Comment (chase) [reply libreoffice-core/sw/source/core/doc/DocumentContentOperationsManager.cxx:4125]: Ah, if that's the case, you can update the comment to reflect it's for both. Just wanted to make sure it wasn't on accident.

**Review by chase — APPROVED** [review]

**Pull request merged**